tests/pytest/enrollment/test_views.py

```diff
@@ -249,6 +249,17 @@ def test_index_eligible_get(client, model_EligibilityType):
     assert "token_field" in response.context_data
     assert "form_retry" in response.context_data
     assert "form_success" in response.context_data
+    assert "overlay_language" in response.context_data
+
+
+@pytest.mark.django_db
+@pytest.mark.usefixtures("mocked_session_agency", "mocked_session_verifier", "mocked_session_eligibility")
+@pytest.mark.parametrize("LANGUAGE_CODE, overlay_language", [("en", "en"), ("es", "es-419")])
+def test_index_eligible_get_changed_language(client, LANGUAGE_CODE, overlay_language):
+    path = reverse(ROUTE_INDEX)
+    response = client.get(path, headers={"accept-language": LANGUAGE_CODE})
```

Comment:
Nice, this works well!

This line made me curious about how Django handles setting and reading the current language, and I thought I'd share what I learned. 

The docs about [How Django discovers language preference](https://docs.djangoproject.com/en/5.0/topics/i18n/translation/#how-django-discovers-language-preference) lists out the various places that the current language could come from, in order of precedence:
 - the URL (doesn't apply to us because we're not using [`i18n_patterns`](https://docs.djangoproject.com/en/5.0/topics/i18n/translation/#how-django-discovers-language-preference) in our [root URLconf](https://github.com/cal-itp/benefits/blob/main/benefits/urls.py#L22))
 - the Django language cookie
 - the `Accept-Language` header sent by the browser
 - the global LANGUAGE_CODE setting

Our `lang-selector.html` template makes use of the built-in [`set_language` view](https://docs.djangoproject.com/en/5.0/topics/i18n/translation/#the-set-language-redirect-view) when the form is [submitted](https://github.com/cal-itp/benefits/blob/main/benefits/core/templates/core/includes/lang-selector.html#L14), so to set the language in a way similar to how a user would, I think we could replace this line with 
```
client.post(reverse("set_language"), data={"language": LANGUAGE_CODE})
response = client.get(path)
```

Either one is perfectly sufficient for the purpose of this test I think, so no change necessary

Reply:
Thanks @angela-tran ! This is good to know, and I like the idea that the test should follow as close as possible how the application is used so I think it's worth it that I make this change 👍